### PR TITLE
Fix DeathSounds not working without DeathTypes defined

### PR DESCRIPTION
**Pull request by abcdefg30**
Needed for #10154.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Sound/DeathSounds.cs [hunk 1]
```diff
@@ -41,11 +41,8 @@ public void Killed(Actor self, AttackInfo e)
 		{
 			var warhead = e.Warhead as DamageWarhead;
 
-			// Killed by some non-standard means
-			if (warhead == null)
-				return;
-
-			if (warhead.DamageTypes.Overlaps(info.DeathTypes))
+			// If the warhead is null, the actor was killed by some non-standard means
+			if (!info.DeathTypes.Any() || (warhead != null && warhead.DamageTypes.Overlaps(info.DeathTypes)))
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Sound/DeathSounds.cs]: `info.DeathTypes.Count == 0` please.

**Comment by abcdefg30** [comment]
Updated.

**Comment by RoosterDragon** [comment]
:+1:

**Comment by obrakmann** [comment]
Seems fine, :+1:

Side question: any idea how to do this nicely for `WithDeathAnimation` when you kill an actor with `Player.Kill()` or `Health.Kill()`, ie. when there's no warhead involved (without breaking crushable)?

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/c4880f0b0604d62a2cfd2ca2168f030ec3a61eed)

**Comment by abcdefg30** [comment]
> Side question: any idea how to do this nicely for  WithDeathAnimation  when you kill an actor with  Player.Kill()  or  Health.Kill() , ie. when there's no warhead involved (without breaking crushable)?

I actually did this in #9350 (https://github.com/abcdefg30/OpenRA/commit/f6e1b570ab8a68a2671f30b8b7fc392effc12d81).
Anyway, good that you reminded me, I cherry-picked that out and filed #10191.

**Pull request merged**